vavr/src/test/java/io/vavr/MatchTest.java

```diff
@@ -302,21 +299,40 @@ public void shouldDecomposeListOfTuple3() {
         assertThat(actual).isEqualTo("(begin, 10, 4.5)::List((middle, 11, 0.0), (end, 12, 1.2))");
     }
 
-    @SuppressWarnings("UnnecessaryLocalVariable")
-    @Test
-    public void shouldDecomposeListWithNonEmptyTail() {
-        final List<Option<Number>> intOptionList = List.of(Option.some(1), Option.some(2.0));
-        final String actual = Match(intOptionList).of(
-                Case($Cons($Some($(1)), $Cons($Some($(2.0)), $())), (x, xs) -> {
-                    // types are inferred correctly!
-                    final Some<Number> head = x;
-                    final List<Option<Number>> tail = xs;
-                    return head + "::" + tail;
-                })
-        );
-        assertThat(actual).isEqualTo("Some(1)::List(Some(2.0))");
-    }
-
+//    @SuppressWarnings("UnnecessaryLocalVariable")
```

Comment:
@danieldietrich, tomorrow is the release date of JDK 9, this commit solves most of the compile problems (I basically recreated everything to validate it, good job, guys (there are two commiters, apparently).
If you could fix these (it's weird generics magic that I didn't have the courage to understand), we could get on the Java 9 wave(r).